### PR TITLE
refactor: use x/oracle in the x/leverage module

### PR DESCRIPTION
**Pull request by alexanderbez**
## Description

- Introduce the `x/oracle` keeper as a dependency on `x/leverage`
- Update oracle/price price related methods in `x/leverage`
- Add `beta` application package for integration and client testing

### Reviewers

- The two core files to review are:
  - `x/leverage/keeper/oracle.go`
  - `x/leverage/keeper/oracle_test.go`
  - `x/leverage/types/expected_types.go`
  - `x/leverage/types/genesis.go`
- There are still changes that need to be made to `x/oracle` to handle the base to display denomination mapping

closes: #263
closes: #97

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -91,12 +91,12 @@ import (
 	appparams "github.com/umee-network/umee/app/params"
 	uibctransfer "github.com/umee-network/umee/x/ibctransfer"
 	uibctransferkeeper "github.com/umee-network/umee/x/ibctransfer/keeper"
-	"github.com/umee-network/umee/x/leverage"
-	leveragekeeper "github.com/umee-network/umee/x/leverage/keeper"
 	leveragetypes "github.com/umee-network/umee/x/leverage/types"
 	"github.com/umee-network/umee/x/peggy"
 	peggykeeper "github.com/umee-network/umee/x/peggy/keeper"
 	peggytypes "github.com/umee-network/umee/x/peggy/types"
+	// "github.com/umee-network/umee/x/leverage"
```
Comment (alexanderbez) [inline app/app.go:98]: We want to disable/not include any incomplete module in the app as we'll be getting ready for an official mainnet release soon.

File: x/leverage/keeper/keeper.go [hunk 2]
```diff
@@ -65,11 +68,6 @@ func (k Keeper) LendAsset(ctx sdk.Context, lenderAddr sdk.AccAddress, loan sdk.C
 		return sdkerrors.Wrap(types.ErrInvalidAsset, loan.String())
 	}
 
-	if !k.bankKeeper.HasBalance(ctx, lenderAddr, loan) {
```
Comment (alexanderbez) [inline x/leverage/keeper/keeper.go:68]: just a note @toteki, there isn't a need to check this as `SendCoinsFromAccountToModule` will error if there is an insufficient balance.

**Comment by alexanderbez** [comment]
This is ready for review -- I will fix the failing tests in the meantime.

File: x/leverage/keeper/oracle.go [hunk 3]
```diff
@@ -7,44 +7,63 @@ import (
...
+	return total, nil
 }
 
 // EquivalentValue returns the amount of a selected denom which would have equal
 // USD value to a provided sdk.Coin
-func (k Keeper) EquivalentValue(ctx sdk.Context, coin sdk.Coin, toDenom string) (sdk.Coin, error) {
-	// get total USD price of input coin (denom, amount)
-	value, err := k.Price(ctx, coin)
+func (k Keeper) EquivalentTokenValue(ctx sdk.Context, fromCoin sdk.Coin, toDenom string) (sdk.Coin, error) {
+	// get total USD price of input (from) denomination
```
Comment (toteki) [inline x/leverage/keeper/oracle.go:54]:
```suggestion
	// get USD price of input (from) denomination
```

Comment (toteki) [inline x/leverage/keeper/oracle.go:54]:
```suggestion
	// get USD price of input (from) denomination
```

File: app/app.go (hunk 1 above)
Comment (alexanderbez) [reply app/app.go:98]: I've added a `beta` package that we can use for integration and client tests in the meantime

**Review by toteki — APPROVED** [review]

**Pull request merged**